src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSupervisorSpec.scala

```diff
@@ -0,0 +1,109 @@
+      new MockGoogleStorageDAO
+    }
+
+    val bucketHelper = new BucketHelper(dataprocConfig, gdDAO, computeDAO, storageDAO, serviceAccountProvider)
+
+    val leoService = new LeonardoService(dataprocConfig, clusterFilesConfig, clusterResourcesConfig,
+      clusterDefaultsConfig, proxyConfig, swaggerConfig, autoFreezeConfig, gdDAO, computeDAO, iamDAO,
+      storageDAO, mockPetGoogleStorageDAO, DbSingleton.ref, whitelistAuthProvider, serviceAccountProvider, whitelist,
+      bucketHelper, contentSecurityPolicy)
+
+    val clusterSupervisorActor = system.actorOf(ClusterMonitorSupervisor.props(monitorConfig, dataprocConfig, clusterBucketConfig, gdDAO,
+      computeDAO, iamDAO, storageDAO, DbSingleton.ref, authProvider, autoFreezeConfig, MockJupyterDAO, MockRStudioDAO, leoService))
+
+    eventually(timeout(Span(30, Seconds))) {
+      val c1 = dbFutureValue { _.clusterQuery.getClusterById(runningCluster.id) }
+      c1.map(_.status) shouldBe Some(ClusterStatus.Stopping)
+    }
+  }
+
+  it should "not auto freeze the cluster if jupyter kernel is still running" in isolatedDbTest {
+    val runningCluster = makeCluster(2).copy(auditInfo = auditInfo.copy(dateAccessed = Instant.now().minus(45, ChronoUnit.SECONDS)),
+      autopauseThreshold = 1).save()
+
+    val clusterRes = dbFutureValue { _.clusterQuery.getClusterById(runningCluster.id) }
+
+    val gdDAO = mock[GoogleDataprocDAO]
+
+    val computeDAO = mock[GoogleComputeDAO]
+
+    val storageDAO = mock[GoogleStorageDAO]
+
+    val iamDAO = mock[GoogleIamDAO]
+
+    val authProvider = mock[LeoAuthProvider]
+
+    val jupyterProxyDAO = new ToolDAO{
+      override def isProxyAvailable(googleProject: GoogleProject, clusterName: ClusterName): Future[Boolean] = Future.successful(true)
+      override def isAllKernalsIdle(googleProject: GoogleProject, clusterName: ClusterName): Future[Boolean] = Future.successful(false)
+    }
+
+    val mockPetGoogleStorageDAO: String => GoogleStorageDAO = _ => {
+      new MockGoogleStorageDAO
+    }
+
+    val bucketHelper = new BucketHelper(dataprocConfig, gdDAO, computeDAO, storageDAO, serviceAccountProvider)
+
+    val leoService = new LeonardoService(dataprocConfig, clusterFilesConfig, clusterResourcesConfig,
+      clusterDefaultsConfig, proxyConfig, swaggerConfig, autoFreezeConfig, gdDAO, computeDAO, iamDAO,
+      storageDAO, mockPetGoogleStorageDAO, DbSingleton.ref, whitelistAuthProvider, serviceAccountProvider, whitelist,
+      bucketHelper, contentSecurityPolicy)
+
+    val clusterSupervisorActor = system.actorOf(ClusterMonitorSupervisor.props(monitorConfig, dataprocConfig, clusterBucketConfig, gdDAO,
+      computeDAO, iamDAO, storageDAO, DbSingleton.ref, authProvider, autoFreezeConfig, jupyterProxyDAO, MockRStudioDAO, leoService))
+
+    eventually(timeout(Span(30, Seconds))) {
+      val c1 = dbFutureValue { _.clusterQuery.getClusterById(runningCluster.id) }
+      c1.map(_.status) shouldNot be (Some(ClusterStatus.Stopped))
```

Comment:
not sure exactly if this is valid check. Would be nice if I can check `c1.map(_.status) shouldBe (Some(ClusterStatus.Running))`. But seems like cluster is not necessarily running  at this point.